core/remote.go

```diff
@@ -51,83 +51,89 @@ func NewRemoteClient(address string) (Client, error) {
 		open:          true,
 		conn:          nothing{},
 	}
-	go client.loop(address)
-	return client, nil
+	return client, client.loop(address)
 }
 
-func (client *remoteSubscriber) loop(address string) {
+func (client *remoteSubscriber) loop(address string) error {
 	// every time we disconnect, we want to reconnect
```

Comment:
Yah that is why inside the for loop it still reconnects. We just found that if a Subscribe was called before the forloop had the opertunity to start the client.conn was nil and caused explosions. For that reason we still have it reconnecting in the for loop but we create a preliminary connection before we allow NewRemoteClient() to return